packages/ui/feature-builder-canvas/src/lib/canvas-utils/drawing/svg-drawer.ts

```diff
@@ -0,0 +1,193 @@
+import { ARC_LENGTH, BUTTON_SIZE, PositionButton, SPACE_BETWEEN_BUTTON_AND_ARROW, SPACE_BETWEEN_VERTICAL_ARROW_LINE_HEIGHT } from './draw-common'
+import { Position } from './step-card'
+
+type SvgOperation = {
+    type: 'M'
+    x: number
+    y: number
+} | {
+    type: 'h' | 'v'
+    delta: number
+} | {
+    type: 'arrow'
+    content: string
+} | {
+    type: 'arc'
+    isRight: boolean
+    content: string
+}
+
+export class SvgDrawer {
+    private readonly operations: SvgOperation[] = []
+
+    constructor(operations: SvgOperation[]) {
+        this.operations = operations
+    }
+
+    static empty(): SvgDrawer {
+        return new SvgDrawer([])
+    }
+
+    private positions(): Position[] {
+        if (this.operations.length === 0) {
+            return []
+        }
+        let currentPositionX = this.operations[0].type === 'M' ? this.operations[0].x : 0
+        let currentPositionY = this.operations[0].type === 'M' ? this.operations[0].y : 0
+        const positions: Position[] = []
+        for (const operation of this.operations) {
+            switch (operation.type) {
+                case 'arc': {
+                    currentPositionX += operation.isRight ? ARC_LENGTH : -ARC_LENGTH
+                    currentPositionY += ARC_LENGTH
+                    break
+                }
+                case 'M': {
+                    currentPositionX = operation.x
+                    currentPositionY = operation.y
+                    break
+                }
+                case 'h': {
+                    currentPositionX += operation.delta
+                    break
+                }
+                case 'v': {
+                    currentPositionY += operation.delta
+                    break
+                }
+                case 'arrow': {
+                    break
+                }
+            }
```

Comment:
Please don't forget to rename these to something meaningful and add add a default case that throws an error which takes a never parameter in it, so we have runtime errors.